module/authentication/lib/src/main/java/org/openbase/bco/authentication/lib/future/AuthenticationFutureList.kt

```diff
@@ -0,0 +1,55 @@
+package org.openbase.bco.authentication.lib.future
+
+import org.openbase.jul.iface.Shutdownable
+import org.openbase.jul.schedule.GlobalScheduledExecutorService
+import org.openbase.jul.schedule.SyncObject
+import org.slf4j.LoggerFactory
+import java.util.concurrent.ExecutionException
+import java.util.concurrent.TimeUnit
+
+object AuthenticationFutureList {
+
+    private const val SCHEDULE_RATE_IN_S = 5L
+    private const val INITIAL_DELAY_IN_S = 1L
+    private const val FUTURE_TIMEOUT_IN_MS = 1L
+
+    private val listSync = SyncObject("AuthenticatedFutureListSync")
+    private val authenticatedFutureList: MutableList<AbstractAuthenticationFuture<*, *>> = ArrayList()
+    private val logger = LoggerFactory.getLogger("AuthenticationFuture")
+
+    private fun isFailed(future: AbstractAuthenticationFuture<*, *>): Boolean {
+        try {
+            // Note: the abstract authentication future will remove itself from this list if
+            //       get finished successfully.
+            future.get(FUTURE_TIMEOUT_IN_MS, TimeUnit.MILLISECONDS)
+        } catch (ex: ExecutionException) {
+            return true
+        }
+        return false
+    }
```

Comment:
How do you want to behave in case an TimeoutException occurs? You want to return `false` right? However, its not handled here and therefore the function will rethrow the TimeoutException which will lead to no further future checks of the other elements in the `authenticatedFutureList`.